sources/api/early-boot-config/src/compression.rs

```diff
@@ -0,0 +1,205 @@
+//! This module supports reading from an input source that could be compressed or plain text.
+//!
+//! Currently gzip compression is supported.
+
+use flate2::read::GzDecoder;
+use std::fs::File;
+use std::io::{BufReader, Chain, Cursor, ErrorKind, Read, Result, Take};
+use std::path::Path;
+
+/// "File magic" that indicates file type is typically two bytes at the start.
```

Comment:
Could we clarify this a little bit more?  Is this mostly true for all file types?

Reply:
It's not always true, for example bzip2 is 3 bytes, but at least a couple other compression/packaging formats are two bytes.  ([reference](https://en.wikipedia.org/wiki/List_of_file_signatures))  I expect this to have to change.  Probably we'd read in the max number of bytes needed for any compression format we care about, and then check the appropriate length prefix of it.  I'll try to clarify the comment.